### PR TITLE
[qa] Moved CI to GitHub Actions

### PR DESCRIPTION
**Pull request by pandafy**

### CODE REVIEW

**Comment by pandafy** [comment]
@Purhan @devkapilbansal can you help me debug why the coverage task is failing?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      run: |
+        ./run-qa-checks
+
+
+    - name: Tests
+      run: |
+        coverage run --source=openwisp_utils runtests.py
+
+    - name: Upload Coverage
+      run: coveralls
```
Comment (devkapilbansal) [inline .github/workflows/ci.yml:60]: Coveralls has been updated, now you need to pass service flag to with it
```suggestion
      run: coveralls --service=github
```

Comment (pandafy) [reply .github/workflows/ci.yml]: Thanks! :+1:

**Review by Purhan — APPROVED** [review]
LGTM :smile: :+1:

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - 3.6
+          - 3.7
+        django-version:
+          - django~=2.2
```
Comment (devkapilbansal) [inline .github/workflows/ci.yml:26]: It would be better to drop `django~=2.2` now.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (atb00ker) [reply .github/workflows/ci.yml:26]: 2.2 is LTS with support till April 2022, check here: https://www.djangoproject.com/download/#supported-versions
Many users will stick to 2.2 for a long time, I think we should drop it in 2022 only! :smile:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    - name: Install openwisp-utils
+      run: |
+        pip install -e .[qa,rest]
+        pip install ${{ matrix.django-version }}
+
+    - name: QA checks
+      run: |
+        ./run-qa-checks
+
+
```
Comment (atb00ker) [inline .github/workflows/ci.yml:54]: Extra space!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        pip install -U -r requirements-test.txt
+
+    - name: Install openwisp-utils
+      run: |
+        pip install -e .[qa,rest]
+        pip install ${{ matrix.django-version }}
+
+    - name: QA checks
+      run: |
+        ./run-qa-checks
```
Comment (atb00ker) [inline .github/workflows/ci.yml:52]: Can't this be `run: ./run-qa-checks` ? :smile:

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+
+    - name: Tests
+      run: |
+        coverage run --source=openwisp_utils runtests.py
```
Comment (atb00ker) [inline .github/workflows/ci.yml:57]: `run: coverage run --source=openwisp_utils runtests.py`

**Review by atb00ker — COMMENTED** [review]
LGTM, minor changes below:

File: .github/workflows/ci.yml (hunk 4 above)
Comment (nemesisdesign) [reply .github/workflows/ci.yml:52]: right, one line is better in this case

File: .github/workflows/ci.yml (hunk 5 above)
Comment (nemesisdesign) [reply .github/workflows/ci.yml:57]: same here

**Review by nemesisdesign — COMMENTED** [review]
LGTM apart minor issues below.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (atb00ker) [reply .github/workflows/ci.yml:26]: Update on this: https://github.com/openwisp/openwisp-radius/pull/175#issuecomment-770219326

**Review by atb00ker — APPROVED** [review]
LGTM! :smile:

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**